packages/protocol/contracts/governance/BlockchainParameters.sol

```diff
@@ -75,6 +101,49 @@ contract BlockchainParameters is Ownable, Initializable {
     emit IntrinsicGasForAlternativeFeeCurrencySet(gas);
   }
 
+  /**
+   * @notice Sets the uptime lookback window.
+   * @param window New window.
+   */
+  function setUptimeLookbackWindow(uint256 window) public onlyOwner {
+    require(window >= 3 && window <= 720, "UptimeLookbackWindow must be within safe range");
+    require(
+      window <= getEpochSize().sub(2),
+      "UptimeLookbackWindow must be smaller or equal to epochSize-2"
+    );
+
+    uptimeLookbackWindow.oldValue = getUptimeLookbackWindow();
+    uptimeLookbackWindow.nextValue = window;
+
+    // epoch 0 is the genesis block, thus implies it has not been initialized
+    if (uptimeLookbackWindow.nextValueActivationEpoch == 0) {
+      // on initialize we set the value for the current epoch which configurer
+      // must make it so it's SAME as geth's chainConfig value
+      uptimeLookbackWindow.nextValueActivationEpoch = getEpochNumber();
```

Comment:
Why is this necessary?  Suppose we always used `getEpochNumber().add(1)`, would that be problematic?

Reply:
For the case where you deploy the contract mid-epoch. 

The change we just did on celo-blockchain helps with that... but only if the contract is already there when donut is activated.